src/widget/controlwidgetconnection.cpp

```diff
@@ -97,40 +97,67 @@ void ControlParameterWidgetConnection::setControlParameterUp(double v) {
 }
 
 ControlWidgetPropertyConnection::ControlWidgetPropertyConnection(
-        WBaseWidget* pBaseWidget, const ConfigKey& key,
-        ValueTransformer* pTransformer, const QString& propertyName)
+        WBaseWidget* pBaseWidget,
+        const ConfigKey& key,
+        ValueTransformer* pTransformer,
+        const QString& propertyName)
         : ControlWidgetConnection(pBaseWidget, key, pTransformer),
-          m_propertyName(propertyName.toLatin1()) {
+          m_propertyName(propertyName) {
+    QWidget* pWidget = pBaseWidget->toQWidget();
+    VERIFY_OR_DEBUG_ASSERT(pWidget) {
+        return;
+    }
+    const QMetaObject* meta = pWidget->metaObject();
+    VERIFY_OR_DEBUG_ASSERT(!propertyName.isEmpty() && meta) {
+        return;
+    }
+    int id = meta->indexOfProperty(propertyName.toLatin1().constData());
+    VERIFY_OR_DEBUG_ASSERT(id >= 0) {
+        return;
+    }
+    m_property = meta->property(id);
+    // skipNoOp = false, during the initial update to synchronize the property in all the sub widgets
     slotControlValueChanged(m_pControl->get());
 }
 
 QString ControlWidgetPropertyConnection::toDebugString() const {
     const ConfigKey& key = getKey();
-    return QString("%1,%2 Parameter: %3 Property: %4 Value: %5").arg(
-            key.group,
-            key.item,
-            QString::number(m_pControl->getParameter()),
-            m_propertyName,
-            m_pWidget->toQWidget()->property(m_propertyName.constData()).toString());
+    return QString("%1,%2 Parameter: %3 Property: %4 Value: %5")
+            .arg(key.group,
+                    key.item,
+                    QString::number(m_pControl->getParameter()),
+                    m_propertyName,
+                    m_property.read(m_pWidget->toQWidget()).toString());
 }
 
 void ControlWidgetPropertyConnection::slotControlValueChanged(double v) {
-    QVariant parameter;
-    QWidget* pWidget = m_pWidget->toQWidget();
-    QVariant property = pWidget->property(m_propertyName.constData());
-    if (property.type() == QVariant::Bool) {
-        parameter = getControlParameterForValue(v) > 0;
+    double parameter = getControlParameterForValue(v);
+    QVariant vParameter;
+    if (m_property.type() == QVariant::Bool) {
+        vParameter = (parameter > 0);
     } else {
-        parameter = getControlParameterForValue(v);
+        vParameter = parameter;
+    }
+    vParameter.convert(m_property.type());
+    if (m_propertyValue == vParameter) {
+        // don't repeat writing the same value that may stall the GUI
+        // Comparing the property value directly does not work, because
+        // in some cases (e.g. visible) it has to be propagated to the child widgets.
+        return;
     }
 
```

Comment:
I'm not sure if this is a good idea. If I understand correctly, this code basically deduplicates subsequent signals which carry the same value. Some GUI controls might rely on the fact that those "duplicate" signals are not swallowed.

Reply:
What are the alternatives? 

One alternative would be to make the no ops updates optional. I have skimmed through the skins and did not find a point where it would be enabled. If we find a point, we can add it.

What do you think?

Reply:
Since this is just influencing the behavior of the CO in the skin and not the CO in general, it should be fine, but this discrepancy/optimization needs to be documented so others don't trip over this when they notice that some CO works differently in Skins and Controllermappings.

Reply:
Good idea. I have add a sentence to https://github.com/mixxxdj/mixxx/wiki/Creating-Skins.